### PR TITLE
Adding Mailjet service

### PR DESCRIPTION
**Pull request by joefearnley**
Adding Mailjet service. Once (if) merged, I will update the README on how to utilize Mailjet.

### CODE REVIEW

**Comment by peter279k** [comment]
I agree with this merge, but some files are conflicted.
It seems that some files does not update the new commit.
Would you use ```git pull``` to solve the conflicts?

Thanks.

**Comment by joefearnley** [comment]
Hi @peter279k, I have updated my forked repo and added back the changes. I will send another pull request. Unfortunately, I do not have the access to pull from the main repo so I had to do it manually. Let me know if I missed anything.

**Comment by peter279k** [comment]
Hi @joefearnley. Did you really try the ```git pull https://github.com/peter279k/serviceBundle``` command?
I think the pull command is will be fine.

Thanks.

**Comment by joefearnley** [comment]
Hi @peter279k, 

I don't think I can pull from your repo and write back without write access. I forked the repo, copied everything over,  and submitted a pull request. If there is anything missing, let me know and I will add it to the repo.

**Comment by peter279k** [comment]
Hi @joefearnley, you can. I fork many repos and pull commit from the orginal repos to synchronize the commits for the many times.
You just try the pull command from original repo and push the latest commit to your forked repo.
And this PR will be updated.

Thanks.

**Comment by peter279k** [comment]
Here is the steps you can refer:
```
git pull https://github.com/peter279k/serviceBundle
# commit or fix the conflicts
git commit -a
git push https://github.com/joefearnley/serviceBundle
```
This will be worked successfully!

**Comment by peter279k** [comment]
Hi @joefearnley, can you remove the ```composer.lock```?
The ```composer.lock``` cached the dependencies cause Travis CI build error.

Thanks.

File: tests/serviceBundle/MailjetTest.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use peter\components\serviceBundle\ServiceFactory;
+
+class MailjetTest extends TestCase
+{
+    /** @test */
+    public function isTypeOfMailgun()
```
Comment (peter279k) [inline tests/serviceBundle/MailjetTest.php]: Please remove the annotation and use the method named ```testIsTypeOfMailjet```.

File: tests/serviceBundle/MailjetTest.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        $mailjetService = (new ServiceFactory)->create('mailjet');
```
Comment (peter279k) [inline tests/serviceBundle/MailjetTest.php]: The ```mailjet``` seems to be wrong.
It should be ```Mailjet```.

File: tests/serviceBundle/MailjetTest.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        $this->assertInstanceOf(peter\components\serviceBundle\Services\Mailjet::class, $mailjetService);
+    }
+
+    /** @test */
+    public function canSendEmail()
```
Comment (peter279k) [inline tests/serviceBundle/MailjetTest.php]: This should remove annotation and use the ```testCanSendEmail``` method name.

File: tests/serviceBundle/MailjetTest.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        $mailjetService->method('sendReq')->willReturn($this->getSuccessReponse());
+
+        $mailjetService->setConfig($config);
+        $response = $mailjetService->sendReq();
+
+        $this->assertSame($response['Messages'][0]['Status'], 'success');
+    }
+
+    /** @test */
+    public function getsErrorWhenCredentialsAreInvalid()
```
Comment (peter279k) [inline tests/serviceBundle/MailjetTest.php]: It should remove the annotation and use the ```testGetsErrorWhenCredentialsAreInvalid``` to be the method name.

**Review by peter279k — CHANGES_REQUESTED** [review]

**Comment by peter279k** [comment]
I write the request changes for this PR.
I notice that the unit testing error during the Travis CI build.
Perhaps you can use the command ```composer test``` after submitting the new commit.

Thanks.

**Comment by joefearnley** [comment]
Hi @peter279k, I think I have all of the code updated per your request. If not, please let me know.

Thanks!

**Comment by peter279k** [comment]
Hi @joefearnley, thank you for your PR.
And you learn how to synchronize the forked repo from the original repo. 

Thank you for your contribution again!

**Comment by joefearnley** [comment]
@peter279k Any time!

**Pull request merged**